code/game/objects/structures/plasticflaps.dm

```diff
@@ -23,14 +23,20 @@
 		return prob(60)
 
 	var/obj/structure/bed/B = A
-	if (istype(A, /obj/structure/bed) && (B.has_buckled_mobs() || B.density))//if it's a bed/chair and is dense or someone is buckled, it will not pass
+	if(istype(A, /obj/structure/bed) && (B.has_buckled_mobs() || B.density))//if it's a bed/chair and is dense or someone is buckled, it will not pass
 		return 0
 
-	if (istype(A, /obj/structure/closet/cardboard))
+	if(istype(A, /obj/structure/closet/cardboard))
 		var/obj/structure/closet/cardboard/C = A
 		if(C.move_delay)
 			return 0
 
+	if(istype(A, /obj/mecha))
+		if(istype(A,/obj/mecha/combat/phazon)) //Since it is a bluespace mech, silly flaps can't stop it.
```

Comment:
Not needed. Just put the mech in Phaze mode and it does not care about **any barriers**.